### DISCUSSION TITLE
Is it possible to run the pre-commit hook before `git amend`?

### DISCUSSION THREAD

**Discussion by pink** [arxanas/git-branchless#1003] [Q&A]
Per title - is there a way to force the pre-commit hook to run when using `amend`?

**Comment by arxanas** [comment]
No, IIRC the `amend` code path creates a commit in-memory and just updates the current `HEAD` to point to it, without any invocation of Git hooks or invocations to `git` itself. You could probably achieve something similar with an alias like `git commit -a --amend && git restack`.
